### PR TITLE
Include number of query cancellations into SHOW POOLS

### PR DESCRIPTION
**Pull request by eulerto**
This commit includes a new metricto SHOW POOLS command: cl_cancel_req.
It informs the number of clients that request query cancellations.

### CODE REVIEW

File: doc/usage.md [hunk 1]
```diff
@@ -365,6 +365,9 @@ cl_active
 cl_waiting
 :   Client connections that have sent queries but have not yet got a server connection.
 
+cl_cancel_req
+:   Client connections that have sent query cancellations.
```
Comment (petere) [inline doc/usage.md:369]: I think this description is a bit misleading. It seems to imply that it is an all-time count of cancel requests that have passed through the pool. But it's actually just the count of cancel requests that have not been forwarded to the server yet, similar to `cl_waiting`. Let's find a better description.

**Review by petere — CHANGES_REQUESTED** [review]

**Comment by eulerto** [comment]
I improved the description and commit message.

**Comment by petere** [comment]
After the recent changes to prioritize cancel requests, is this still necessary? It will be almost impossible to see anything but zero in this field now.

**Comment by eulerto** [comment]
Fair point. However, since it is now limited to 2 x pool size, I'm afraid that a small pool reaches its limit.

Another idea is to add a WARNING to cover these cases (small pool and high number of query cancellations) such as:

```c
  /* when a cancel request is queued allow connections up to twice the pool size */
  if (!statlist_empty(&pool->cancel_req_list) && max < (2 * pool->db->pool_size)) {
    log_debug("launch_new_connection: bypass pool limitations for cancel request");
    goto force_new;
  }
  else if (!statlist_empty(&pool->cancel_req_list)) {
    log_warning("pool %s: query cancellation requests were queued because the connection limit per pool was reached", pool->db->name);
  }
```

We might improve this message informing the `2 * pool size`.

**Comment by jcoleman** [comment]
In my opinion the fact that various bug fixes have been committed around this list is orthogonal to whether this gets merged -- those issues would have been far easier to diagnose, I believe, with this feature added. So it seems to me that it's worth having to be able to expose errors if more are found in the future.

**Pull request merged**